### PR TITLE
[http] Support chaining transformations without an intersection symbol

### PR DESCRIPTION
**Pull request by jimtng**
Depends on https://github.com/openhab/openhab-core/pull/4353 (merged)

### CODE REVIEW

File: bundles/org.openhab.binding.http/README.md [hunk 1]
```diff
@@ -183,10 +185,34 @@ http://www.domain.org/home/lights/23871/?status=OFF&date=2020-07-06
...
+
+Multiple transformations can be chained in the .things file by listing them as comma-separated values:
+
+```java
+Thing http:url:foo "Foo" [
+    baseURL="https://example.com/api/v1/metadata-api/web/metadata",
+    headers="key1=value1", "key2=value2", "key3=value3",
+    refresh=15] {
+        Channels:
+            Type string : text "Text" [ stateTransformation="REGEX(.*meta.*)", "JSONPATH($.metadata.data)" ]
```
Comment (J-N-K) [inline bundles/org.openhab.binding.http/README.md:204]: Is this syntax also working if you configure `stateTransformation` and `commandTransformation`?

Comment (jimtng) [reply bundles/org.openhab.binding.http/README.md:204]: Yes, although I haven't personally tested this on an actual installation.

It should look like this:

```java
  Type string : text "Text" [ 
    stateTransformation="REGEX(.*meta.*)", "JSONPATH($.metadata.data)",
    commandTransformation="REGEX(.*meta.*)", "JSONPATH($.metadata.data)"
   ]
```

File: bundles/org.openhab.binding.http/README.md [hunk 2]
```diff
@@ -71,13 +71,15 @@ The relevant transformation service needs to be installed via the Main UI or add
...
-| `THEVALUE:23.2°C`                                                   | REGEX       | `REGEX::(.*?)°`                           |
+| Received value                                                      | Tr. Service | Transformation                             |
+|---------------------------------------------------------------------|-------------|--------------------------------------------|
+| `{device: {status: { temperature: 23.2 }}}`                         | JSONPATH    | `JSONPATH($.device.status.temperature)`    |
+| `<device><status><temperature>23.2</temperature></status></device>` | XPath       | `XPath(/device/status/temperature/text())` |
+| `THEVALUE:23.2°C`                                                   | REGEX       | `REGEX(:(.*?)°)`                           |
 
-Transformations can be chained by separating them with the mathematical intersection character "∩".
+Transformations can be chained in the UI by listing each transformation on a separate line, or by separating them with the mathematical intersection character "∩".
+Starting from openHAB 4.3, transformations are defined using this syntax: `TYPE(FUNCTION)`, e.g.: `JSONPATH($.path)`.
```
Comment (florian-h05) [inline bundles/org.openhab.binding.http/README.md:81]: I don‘t think we should mention 4.3 here — these doc changes will only be on next.openhab.org until the 4.3 release.

Comment (jimtng) [reply bundles/org.openhab.binding.http/README.md:81]: What should it say instead?

Comment (florian-h05) [reply bundles/org.openhab.binding.http/README.md:81]: I would go for:

> Transformations are defined using this syntax: …
> The syntax `TYPE:FUNCTION` is still supported, e.g. …

Comment (jimtng) [reply bundles/org.openhab.binding.http/README.md]: Thanks, updated!

File: bundles/org.openhab.binding.http/src/main/resources/OH-INF/config/config.xml [hunk 3]
```diff
@@ -5,13 +5,15 @@
 	xsi:schemaLocation="https://openhab.org/schemas/config-description/v1.0.0 https://openhab.org/schemas/config-description-1.0.0.xsd">
 
 	<config-description uri="channel-type:http:channel-config">
-		<parameter name="stateTransformation" type="text">
+		<parameter name="stateTransformation" type="text" multiple="true">
 			<label>State Transformation</label>
-			<description>Transformation pattern used when receiving values. Multiple transformation can be chained using "∩".</description>
+			<description>Transformation pattern used when receiving values. Multiple transformations can be chained by listing
+				each transformation in a separate line, or by concatenating them with "∩".</description>
```
Comment (J-N-K) [inline bundles/org.openhab.binding.http/src/main/resources/OH-INF/config/config.xml:11]: In `README.md` you use `listing ON a separate line`.  Is it intentional to use `listing IN a separate line` here?

Comment (jimtng) [reply bundles/org.openhab.binding.http/src/main/resources/OH-INF/config/config.xml]: I've updated them to "on a separate line" but my grammar isn't the best.

**Comment by jimtng** [comment]
The core functionality that this PR needs has been merged, and I've briefly tested the chaining functionality and it works.

**Pull request merged**